quantum/impl/quantum_task_impl.h

```diff
@@ -106,9 +108,39 @@ int Task::run()
         {
             return (int)ITask::RetCode::Sleeping;
         }
+        CoroutineState coroutineState = CoroutineState::Resumed;
+        if (_isNew)
+        {
+            coroutineState = CoroutineState::Constructed;
```

Comment:
I think maybe we need to call stateHandler with **Constructed** state alongside with **Resumed**

Reply:
I think it makes sense to set the Resumed-state only if the task was previously Suspended. I.e. the state machine should be easy to justify, e.g.
Constructed -> [Suspended -> Resumed]* -> Destructed.

Reply:
Done